src/scene/Scale.js

```diff
@@ -187,8 +193,34 @@ function ordinal(scale, rng, group) {
     };
   }
 
-  prev.range = rng;
-  this._updated = true;
+  // Ordinal scale inversion requires range in ascending order.
+  // Shuffle domain/range to ensure this is true, and cache.
+  if (this._updated) {
```

Comment:
`this._updated` is set to `true` above. I don't see how it could ever be `false` at this point. So why the `if` statement?